src/goto-harness/recursive_initialization.cpp

```diff
@@ -136,5 +151,180 @@ void recursive_initializationt::initialize_nondet(
   const recursion_sett &known_tags,
   code_blockt &body)
 {
-  body.add(code_assignt{lhs, side_effect_expr_nondett{lhs.type()}});
+  if(
+    lhs.id() != ID_symbol ||
+    !is_array_size_parameter(to_symbol_expr(lhs).get_identifier()))
+  {
+    body.add(code_assignt{lhs, side_effect_expr_nondett{lhs.type()}});
+  }
+}
+
+void recursive_initializationt::initialize_array(
+  const exprt &array,
+  std::size_t depth,
+  const recursion_sett &known_tags,
+  code_blockt &body)
+{
+  PRECONDITION(array.type().id() == ID_array);
+  const auto &array_type = to_array_type(array.type());
+  const auto array_size =
+    numeric_cast_v<std::size_t>(to_constant_expr(array_type.size()));
+  for(std::size_t index = 0; index < array_size; index++)
+  {
+    initialize(
+      index_exprt(array, from_integer(index, size_type())),
+      depth,
+      known_tags,
+      body);
+  }
+}
+
+bool recursive_initializationt::should_be_treated_as_array(
+  const irep_idt &array_name) const
+{
+  return initialization_config.pointers_to_treat_as_arrays.find(array_name) !=
+         initialization_config.pointers_to_treat_as_arrays.end();
```

Comment:
⛏ `set::count` is shorter. You could even get away without this method - especially since it's used only once.
```c++
initialization_config.pointers_to_treat_as_arrays.count(array_name);
```

Reply:
The reason these are in named functions is twofold:
1. The logic for these will get more complicated in the future if we decide to support more complex scenarios
2. This is spelling out what it's intended to do, rather than just showing what it _does_ do

Reply:
You still can use `count` though.